src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -493,13 +493,13 @@ public function getCompiler()
      * @param string $id      The service identifier
      * @param object $service The service instance
      *
-     * @throws BadMethodCallException When this ContainerBuilder is frozen
+     * @throws BadMethodCallException When this ContainerBuilder is compiled
      */
     public function set($id, $service)
     {
         $id = $this->normalizeId($id);
 
-        if ($this->isFrozen() && (isset($this->definitions[$id]) && !$this->definitions[$id]->isSynthetic())) {
+        if ($this->isCompiled() && (isset($this->definitions[$id]) && !$this->definitions[$id]->isSynthetic())) {
             // setting a synthetic service on a frozen container is alright
             throw new BadMethodCallException(sprintf('Setting service "%s" for an unknown or non-synthetic service definition on a frozen container is not allowed.', $id));
```

Comment:
`sprintf('Setting service "%s" for an unknown or non-synthetic service definition on a compiled container is not allowed.', $id)`

basically, `s/frozen/compiled`, no ?